prov/verbs/src/fi_verbs.h

```diff
@@ -465,16 +465,15 @@ struct fi_ibv_srq_ep {
 	struct ibv_srq		*srq;
 	struct fi_ibv_domain	*domain;
 
-	/* Due to XRC SRQ semantics, an XRC SRX context may only be bound
-	 * to multiple endpoints when they share the same RX CQ. */
-	struct fi_ibv_cq	*srq_cq;
-
-	/* XRC SRQ is not created until endpoint enable */
-	fastlock_t		prepost_lock;
-	struct slist		prepost_list;
-	uint32_t		max_recv_wr;
-	uint32_t		max_sge;
-	uint32_t		prepost_count;
+	/* For XRC SRQ only */
+	struct {
+		/* XRC SRQ is not created until endpoint enable */
+		fastlock_t		prepost_lock;
+		struct slist		prepost_list;
+		uint32_t		max_recv_wr;
+		uint32_t		max_sge;
```

Comment:
Why are max_recv_wr and max_sge needed?  I would expect the app to adhere to the limits returned from fi_getinfo.

Reply:
The XRC SRQ can not be created when the SRQ endpoint is created; it needs the the RX CQ from the EP before it can be created. These fields are taken from the RX attributes specified when the SRQ endpoint is created, and then used later at EP enable to control the sizing of XRC SRQ created.

Reply:
max_recv_wr is also used to validate XRC SRQ pre-posts. The only Verbs software detected errors are SRQ overflow and/or WQE overflow and these allow an error to be returned at the time of the pre-post. The remaining errors will be returned through CQ/events.